codeguru_profiler_agent/sdk_reporter/sdk_reporter.py

```diff
@@ -90,11 +113,55 @@ def report(self, profile):
             )
             logger.info("Reported profile successfully")
             return True
+        except ClientError as error:
+            if error.response['Error']['Code'] == 'ResourceNotFoundException':
+                if self.is_compute_platform_lambda():
+                    self.__class__.is_create_pg_called_during_submit_profile = True
+                    logger.info(
+                        "Profiling group not found. Will try to create a profiling group"
+                        "with name = {} and compute platform = {}".format(self.profiling_group_name, 'AWSLambda'))
+                    self.create_profiling_group()
+            return False
         except Exception as e:
             self._log_request_failed(operation="post_agent_profile", exception=e)
             return False
 
+    @with_timer("createProfilingGroup", measurement="wall-clock-time")
+    def create_profiling_group(self):
+        """
+        Create a Profiling Group for the AWS Lambda function.
+        """
+        try:
+            self.codeguru_client_builder.codeguru_client.create_profiling_group(
+                profilingGroupName=self.profiling_group_name,
+                computePlatform='AWSLambda'
+            )
+            self.is_profiling_group_created_during_execution = True
```

Comment:
I don't see any usage of `is_profiling_group_created_during_execution `. Can you explain why are we having this?

Reply:
Yeah was using it before. Removed it!